### ISSUE TITLE
Dashboards: Snapshot "Open original dashboard" button ignores subpath

### ISSUE THREAD

**Issue by crinjes** [grafana/grafana#76770]
### What happened?

When clicking the "Open original dashboard" button on a snapshot, the subpath in the URL is lost.

With a root_url of `http://localhost:3000/grafana/` and `serve_from_sub_path = true`, the button will attempt to open `http://localhost:3000/d/...` instead of `http://localhost:3000/grafana/d/...`

### What did you expect to happen?

The subpath should be included in the URL.

### Did this work before?

Unknown

### How do we reproduce it?

1. `docker run --name grafana --rm -p 3000:3000  -e "GF_SERVER_DOMAIN=localhost" -e "GF_SERVER_ROOT_URL=%(protocol)s://%(domain)s:%(http_port)s/grafana/" -e "GF_SERVER_SERVE_FROM_SUB_PATH=true" grafana/grafana-oss:main`
2. sign in at http://localhost:3000/grafana/, create a dashboard, create a local snapshot, and open the snapshot URL
3. click the "Open original dashboard" button while observing the browser's HTTP requests

The request will go to `http://localhost:3000/d/...`. The builtin server masks this error by 301 redirecting to the correct `http://localhost:3000/grafana/d/...`

### Environment (with versions)?

Grafana: 10.1.2, reproduced with `grafana/grafana-oss:main` container, commit 91ad07719f version 10.3.0-141747
OS: Windows
Browser: Chrome, Firefox

### Grafana platform?

Docker

**Comment by usmangt** [comment]
Hi @crinjes,

I am closing this issue here on GitHub, as it seems more of a question.

Please create a new post at our community forums in the [Configuration category](https://community.grafana.com/c/support/configuration/20) by providing the same details (in case if you missed any, provide that as well) so that our community members can help you out.

Thanks

**Comment by crinjes** [comment]
@usmangt No, this is a bug. The "Open original dashboard" button on a snapshot uses an invalid URL in its redirect when using subpaths.
